### ISSUE TITLE
Supports let-chains in `manual_strip`

### ISSUE THREAD

**Issue by ada4a** [rust-lang/rust-clippy#15818]
Currently, the lint doesn't flag code like this:
```rs
#![warn(clippy::manual_strip)]

fn some_check() -> bool { todo!() }

fn foo(s: &str) {
    if s.starts_with("ab")
        && let stripped = &s["ab".len()..]
        && some_check() // to avoid `irrefutable_let_patterns`
    {
        //
    }
}
```
It would be cool if it would.

I think checking only consecutive `if`s is fine.

@rustbot label C-enhancement
